drools-ansible-rulebook-integration-main/src/main/java/org/drools/ansible/rulebook/integration/main/Payload.java

```diff
@@ -29,61 +23,61 @@ private Payload(List<String> list) {
         List<String> payloadList = new ArrayList<>();
 
         int repeatCount = 1;
         try {
-            repeatCount = sourcesArgs.getInt("repeat_count");
-        } catch (JSONException e) { /* ignore */ }
+            repeatCount = Integer.valueOf(sourcesArgs.get("repeat_count").toString());
+        } catch (Exception e) { /* ignore */ }
 
         try {
             for (int i = 0; i < repeatCount; i++) {
-                for (Object p : (JSONArray) sourcesArgs.get("payload")) {
-                    payloadList.add(p.toString());
+                for (Object p : (List) sourcesArgs.get("payload")) {
+                    payloadList.add(JsonMapper.toJson(p));
                 }
             }
-        } catch (JSONException e) { /* ignore */ }
+        } catch (Exception e) { /* ignore */ }
 
         try {
-            String indexName = sourcesArgs.getString("create_index");
+            String indexName = sourcesArgs.get("create_index").toString();
             for (int i = 0; i < repeatCount; i++) {
                 String payload = "{\"" + indexName + "\" : " + i + "}";
                 payloadList.add(payload);
             }
-        } catch (JSONException e) { /* ignore */ }
+        } catch (Exception e) { /* ignore */ }
 
         Payload payload = new Payload(payloadList);
 
         try {
-            payload.eventDelay = sourcesArgs.getInt("delay");
-        } catch (JSONException e) {
+            payload.eventDelay = Integer.valueOf(sourcesArgs.get("delay").toString());
+        } catch (Exception e) {
             try {
-            payload.eventDelay = sourcesArgs.getInt("event_delay");
-            } catch (JSONException e1) {
+            payload.eventDelay = Integer.valueOf(sourcesArgs.get("event_delay").toString());
+            } catch (Exception e1) {
                 /* ignore */
             }
         }
         try {
-            payload.startDelay = sourcesArgs.getInt("start_delay");
-        } catch (JSONException e) {
+            payload.startDelay = Integer.valueOf(sourcesArgs.get("start_delay").toString());
+        } catch (Exception e) {
             /* ignore */
         }
         try {
-            payload.loopCount = sourcesArgs.getInt("loop_count");
-        } catch (JSONException e) {
+            payload.loopCount = Integer.valueOf(sourcesArgs.get("loop_count").toString());
+        } catch (Exception e) {
             /* ignore */
         }
         try {
-            payload.loopDelay = sourcesArgs.getInt("loop_delay");
-        } catch (JSONException e) {
+            payload.loopDelay = Integer.valueOf(sourcesArgs.get("loop_delay").toString());
+        } catch (Exception e) {
             /* ignore */
         }
         try {
-            payload.shutdown = sourcesArgs.getInt("loop_delay");
-        } catch (JSONException e) {
+            payload.shutdown = Integer.valueOf(sourcesArgs.get("loop_delay").toString());
+        } catch (Exception e) {
```

Comment:
Do we really need to catch all exceptions? I would prefer a Jackson specific one if possible or even multiple exceptions if necessary.
Same comment for all the usages in this class

Reply:
can you elaborate on _why_ prefer specific ones, given in _any_ case the net result is the field is not set.
in the meantime I will update anyway

Reply:
No reasons in particular more than I prefer to avoid a generic catch `Exception` for similar "parsing" issues.
Especially in a similar migration PR it makes sure to avoid hiding a bug/regression (I don't know how good is the current coverage of this code)